pkg/odo/cli/component/create.go

```diff
@@ -423,10 +433,61 @@ func (co *CreateOptions) Run() (err error) {
 	if err != nil {
 		return errors.Wrapf(err, "failed to persist the component settings to config file")
 	}
-	log.Infof("Please use `odo push` command to create the component with source deployed")
+	if co.now {
+		co.Context, co.localConfigInfo, err = genericclioptions.UpdatedContext(co.Context)
+
+		if err != nil {
+			return errors.Wrap(err, "unable to retrieve updated local config")
+		}
+		err = co.SetSourceInfo()
+		if err != nil {
+			return errors.Wrap(err, "unable to set source information")
+		}
+		err = co.Push()
+		if err != nil {
+			return errors.Wrapf(err, "failed to push the changes")
+		}
+	} else {
+		log.Infof("Please use `odo push` command to create the component with source deployed")
```

Comment:
is it me or are we missing the `\n` here? not too sure.

Reply:
Updating